java/java-ranger-regression/alarm/prop2_True/Main.java

```diff
@@ -0,0 +1,426 @@
+        rtu_const_in.System_Test_Interval = System_Test_Interval;
+        rtu_const_in.Max_Display_Duration = Max_Display_Duration;
+        rtu_const_in.Max_Confirm_Stop_Duration = Max_Confirm_Stop_Duration;
+
+
+        System_Status_Outputs rtu_sys_stat_in = new System_Status_Outputs();
+        rtu_sys_stat_in.Reservoir_Empty = Reservoir_Empty;
+        rtu_sys_stat_in.Reservoir_Volume = Reservoir_Volume1;
+        rtu_sys_stat_in.Volume_Infused = Volume_Infused;
+        rtu_sys_stat_in.Log_Message_ID = Log_Message_ID3;
+        rtu_sys_stat_in.In_Therapy = In_Therapy;
+
+        Config_Outputs rtu_config_in = new Config_Outputs();
+        rtu_config_in.Patient_ID = Patient_ID;
+        rtu_config_in.Drug_Name = Drug_Name2;
+        rtu_config_in.Drug_Concentration = Drug_Concentration;
+        rtu_config_in.Infusion_Total_Duration = Infusion_Total_Duration;
+        rtu_config_in.VTBI_Total = VTBI_Total;
+        rtu_config_in.Flow_Rate_Basal = Flow_Rate_Basal;
+        rtu_config_in.Flow_Rate_Intermittent_Bolus = Flow_Rate_Intermittent_Bolus;
+        rtu_config_in.Duration_Intermittent_Bolus = Duration_Intermittent_Bolus;
+        rtu_config_in.Interval_Intermittent_Bolus = Interval_Intermittent_Bolus;
+        rtu_config_in.Flow_Rate_Patient_Bolus = Flow_Rate_Patient_Bolus;
+        rtu_config_in.Duration_Patient_Bolus = Duration_Patient_Bolus;
+        rtu_config_in.Lockout_Period_Patient_Bolus = Lockout_Period_Patient_Bolus;
+        rtu_config_in.Max_Number_of_Patient_Bolus = Max_Number_of_Patient_Bolus;
+        rtu_config_in.Flow_Rate_KVO = Flow_Rate_KVO2;
+        rtu_config_in.Entered_Reservoir_Volume = Entered_Reservoir_Volume;
+        rtu_config_in.Reservoir_Volume = Reservoir_Volume2;
+        rtu_config_in.Configured = Configured;
+        rtu_config_in.Error_Message_ID = Error_Message_ID;
+        rtu_config_in.Request_Config_Type = Request_Config_Type;
+        rtu_config_in.Request_Confirm_Infusion_Initiate = Request_Confirm_Infusion_Initiate;
+        rtu_config_in.Request_Patient_Drug_Info = Request_Patient_Drug_Info;
+        rtu_config_in.Request_Infusion_Info = Request_Infusion_Info;
+        rtu_config_in.Log_Message_ID = Log_Message_ID4;
+        rtu_config_in.Config_Timer = Config_Timer;
+        rtu_config_in.Config_Mode = Config_Mode;
+
+
+        Alarm_Outputs rty_alarm_out = new Alarm_Outputs();
+        rty_alarm_out.Is_Audio_Disabled = Is_Audio_Disabled;
+        rty_alarm_out.Notification_Message = Notification_Message;
+        rty_alarm_out.Audio_Notification_Command = Audio_Notification_Command;
+        rty_alarm_out.Highest_Level_Alarm = Highest_Level_Alarm;
+        rty_alarm_out.Log_Message_ID = Log_Message_ID5;
+
+        B_ALARM_Functional_c_T localB = new B_ALARM_Functional_c_T();
+        DW_ALARM_Functional_f_T localDW = new DW_ALARM_Functional_f_T();
+
+        alarm.ALARM_Functional_Init(localB, localDW);
+
+        if ((0 <= Commanded_Flow_Rate) &&
```

Comment:
Could you elaborate on the need for bounds? Where is the problem, if you start with 256 compared to 255?
Might it be preferable initializing the values with `Random.nextInt(255);` instead of `Verifier.nondetInt()` and this very long if statement?


This code seems to model a uint8 value range, but this is not going to happen during mathematical operations. Is this a problem for the verification task?

Reply:
The bounds are needed to model uint8. I have shortened the if-statement considerably to constraint bounds on important/used variables.